### ISSUE TITLE
请问一下训练文时引入的tqdm是什么文件

### ISSUE THREAD

**Issue by huyi1989** [lc222/seq2seq_chatbot_new#2]
from tqdm import tqdm ？？

**Comment by lc222** [comment]
就是python的一个库，用来显示程序运行时间的，你直接pip安装tqdm

**Comment by huyi1989** [comment]
好的，谢谢了。我这里还有一个问题想请教一下，我将你的代码修改了一下，想要训练一个自编码器，就是输出等于输入，但是我在运行训练的时候出现了错误，报的错误是这样的：InvalidArgumentError (see above for traceback): logits and labels must have the same first dimension，是说解码训练计算loss的时候，logits和labels它们的长度不一致。我猜想是因为输入的数据是经过padding的，同时输入了对应的序列长度，而输出的结果是计算到给定最大长度的句子为止，这里可能产生错误。想请教一下知道怎么解决这个错误吗？

**Comment by lc222** [comment]
针对你的这个猜想，你可以将输入数据Padding的长度跟橘子的最大长度设成相同的值。并且要保证给定的最大长度大于所有的训练样本，否则将大的训练样本进行截断操作。另外，我看报错说第一个纬度不同，是不是因为二者的batch_size不同？？具体的还要结合你的代码看。

**Comment by huyi1989** [comment]
好的，谢谢，问题解决了，原因是在数据预处理的时候，我取了所有数据中最大长度的句子作为最大长度，导致最大长度不是随着每个batch变化的，所以会出现长度对不齐。
